packages/lib/src/components/PreAuthorizedDebitCanada/PreAuthorizedDebitCanada.tsx

```diff
@@ -0,0 +1,101 @@
+import { h, Fragment } from 'preact';
+import UIElement from '../internal/UIElement/UIElement';
+import { CoreProvider } from '../../core/Context/CoreProvider';
+import { TxVariants } from '../tx-variants';
+import PreAuthorizedDebitCanadaComponent from './components/PreAuthorizedDebitCanadaComponent';
+import { SettlementInfo } from './components/SettlementInfo';
+import RedirectButton from '../internal/RedirectButton';
+import { payAmountLabel } from '../internal/PayButton';
+
+import type { PreAuthorizedDebitCanadaConfiguration } from './types';
+
+export class PreAuthorizedDebitCanada extends UIElement<PreAuthorizedDebitCanadaConfiguration> {
+    public static type = TxVariants.eft_directdebit_CA;
+
+    public override formatData() {
+        const recurringPayment = !!this.props.storedPaymentMethodId;
+
+        if (recurringPayment) {
+            return {
+                paymentMethod: {
+                    type: PreAuthorizedDebitCanada.type,
+                    storedPaymentMethodId: this.props.storedPaymentMethodId
+                }
+            };
+        }
+
+        return {
+            paymentMethod: {
+                type: PreAuthorizedDebitCanada.type,
+                ownerName: this.state.data.ownerName,
+                bankAccountNumber: this.state.data.bankAccountNumber,
+                bankCode: this.state.data.bankCode,
+                bankLocationId: this.state.data.bankLocationId
+            },
+            ...(this.state.storePaymentMethod && { storePaymentMethod: this.state.storePaymentMethod })
+        };
+    }
+
+    public override formatProps(props: PreAuthorizedDebitCanadaConfiguration): PreAuthorizedDebitCanadaConfiguration {
+        return {
+            enableStoreDetails: props.session?.configuration?.enableStoreDetails ?? props.enableStoreDetails,
```

Comment:
Maybe this `enableStoreDetails` logic could be lifted up to UIElement. Every payment that has tokenization needs to have this logic in place, and we might forget to add it